crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -103,7 +105,46 @@ pub trait EthCall: EstimateCall + Call + LoadPendingBlock + LoadBlock + FullEthA
                 let mut gas_used = 0;
                 let mut blocks: Vec<SimulatedBlock<RpcBlock<Self::NetworkTypes>>> =
                     Vec::with_capacity(block_state_calls.len());
+
+                let block_state_calls = iter::from_fn({
+                    let mut iter = block_state_calls.into_iter();
+                    let mut prev = None;
+
+                    move || {
+                        let block = iter.next()?;
+
+                        let this = block.block_overrides.as_ref().and_then(|b| b.number);
+
+                        let range = iter::successors(prev, |p| U256::checked_add(*p, U256::ONE));
+                        let range = this
+                            .map(|t| range.skip(1).take_while(move |i| i < &t))
+                            .into_iter()
+                            .flatten();
+                        prev = this;
+
+                        let gaps = range.map(|n| {
+                            SimBlock::default().with_block_overrides(BlockOverrides {
+                                number: Some(n),
+                                ..Default::default()
+                            })
+                        });
+
+                        Some(gaps.chain(Some(block)))
+                    }
+                })
```

Comment:
I think this looks a bit too complex for what this should do.

we can likely solve this easier by creating a new vec of `sanitized_simblocks` and we iterate over the provided args and fill gaps with empty block args